### PR TITLE
mgr/cephadm: fix host drain with agent

### PR DESCRIPTION
**Pull request by adk3798**
Agent was not getting removed from hosts with _no_schedule label
since it was using all hosts rather than _schedulable hosts. Added
a _non_draining hosts function to return a prope list of hosts agent
is okay to schedule on.

Was stuck between using a daemon to report which daemons are on the
host and wanting to remove all the daemons on the host when draining.
Now using ssh to confirm all the daemons have left the host.

Signed-off-by: Adam King <adking@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -1391,6 +1391,18 @@ def _schedulable_hosts(self) -> List[HostSpec]:
             )
         ]
 
+    def _non_draining_hosts(self) -> List[HostSpec]:
```
[the remark this answers is not in the document]
Comment (adk3798) [reply src/pybind/mgr/cephadm/module.py:1394]: I like it. Do we want to migrate _schedulable_hosts and _unreachable_hosts as well? I could see them all just existing as class functions for the host cache similar to how we have a variety of function to get daemons with certain criteria.

[the remark this answers is not in the document]
Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py:1394]: +1

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Review by nizamial09 — APPROVED** [review]
tested locally! LGTM

**Comment by adk3798** [comment]
http://pulpito.front.sepia.ceph.com/adking-2021-10-01_02:06:07-rados:cephadm-wip-adk-testing-2021-09-30-1743-distro-basic-smithi/

One failure tracked by https://tracker.ceph.com/issues/49287
Other failure was a failed install command, looks unrelated

`Command failed on smithi114 with status 100: 'sudo DEBIAN_FRONTEND=noninteractive apt-get -y --force-yes -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" install ceph=17.0.0-8016-g98a17fdb-1focal cephadm=17.0.0-8016-g98a17fdb-1focal ceph-mds=17.0.0-8016-g98a17fdb-1focal ceph-mgr=17.0.0-8016-g98a17fdb-1focal ceph-common=17.0.0-8016-g98a17fdb-1focal ceph-fuse=17.0.0-8016-g98a17fdb-1focal ceph-test=17.0.0-8016-g98a17fdb-1focal ceph-volume=17.0.0-8016-g98a17fdb-1focal radosgw=17.0.0-8016-g98a17fdb-1focal python3-rados=17.0.0-8016-g98a17fdb-1focal python3-rgw=17.0.0-8016-g98a17fdb-1focal python3-cephfs=17.0.0-8016-g98a17fdb-1focal python3-rbd=17.0.0-8016-g98a17fdb-1focal libcephfs2=17.0.0-8016-g98a17fdb-1focal libcephfs-dev=17.0.0-8016-g98a17fdb-1focal librados2=17.0.0-8016-g98a17fdb-1focal librbd1=17.0.0-8016-g98a17fdb-1focal rbd-fuse=17.0.0-8016-g98a17fdb-1focal'`

**Pull request merged**